### PR TITLE
Update SmallRye Config to 1.11.0

### LINKED ISSUE #9895: Allow activation of multiple configuration profiles
**Description**
I understand that currently a Quarkus config profile denotes an "environment".

But sometimes you want to resuse configuration bits but you cannot put them in the "common" block (properties without `%<profileName>` prefix).
E.g. in my current project we would like to toggle dev-mode between h2 (default) and postgres (also used in prod, but with slightly different config). We have some more dev specific settings like logging to a file (which we don't do in prod) etc.
In this scenario we have to duplicate many properties from `dev` to `dev-postgres` which violates the DRY principle and makes the config less readable.
(To make things worse we have multiple "layers" of config files in various submodules which are aggregated via a custom `ConfigSourceProvider`.)

It would come in handy if multiple profiles could be actived.

**Implementation ideas**
Allow profiles to be set in a comma separated way, e.g. `foo,bar` (`bar` would overwrite `foo`, Spring Boot style IIRC).

Fun fact (or better to say bug): As of 1.4.2.Final you _can_ set `-Dquarkus.profile=foo,bar` and Quarkus will happily report that it activated `foo,bar`, but nothing of the two profiles is taken into account.

### LINKED ISSUE #11475: Optional nested configuration issue
**Describe the bug**
I have a Bean annotated with @ConfigProperties with a nested element; which seems to work. However, if I make that nested element optional, I get a missing converter exception.

**Expected behavior**
I would expect that an optional element would not need to be defined in my application.yml, but if the nested element is defined then I would expect the validation within that nested element would apply to those fields.

**Actual behavior**
On startup, I get the following exception:
 java.lang.IllegalArgumentException: SRCFG00013: No Converter registered for class dev.buesing.config.Cluster$SslEmbedded

**To Reproduce**
I have a sample project I can include to show the error

1. Create a bean with @ConfigProperties
2. Add a nested element making it Optional<>, e.g. Optional<Ssl>
3. Do not include any entry in the application.yml for that optional element
4. start the application

Cluster (with the SslEmbedded static inner class)
```
package dev.buesing.config;

import io.quarkus.arc.config.ConfigProperties;
import lombok.Data;

import javax.validation.constraints.Min;
import java.util.List;
import java.util.Optional;

@Data
@SuppressWarnings("unused")
@ConfigProperties(prefix = "cluster")
public class Cluster {

    @Data
    @SuppressWarnings("unused")
    public static class SslEmbedded {
        private String truststoreLocation;
        private String truststorePassword;
        private String keystoreLocation;
        private String keystorePassword;
        private String keyPassword;
    }

    private List<String> bootstrapServers;
    private Optional<List<String>> schemaRegistryServers;
    private String securityProtocol = "SSL";
    private Optional<Ssl> ssl;
    private Optional<SslEmbedded> sslEmbedded;

    @Min(1)
    private Integer linger;
}
```

Ssl (separate class)
```
package dev.buesing.config;

import lombok.Data;

@Data
@SuppressWarnings("unused")
public class Ssl {
    private String truststoreLocation;
    private String truststorePassword;
    private String keystoreLocation;
    private String keystorePassword;
    private String keyPassword;
}
```

build.gradle
```
plugins {
    id 'idea'
    id 'io.quarkus' version '1.7.0.Final' apply false
}


repositories {
    mavenLocal()
    mavenCentral()
    jcenter()
}

version = '1.0'


apply plugin: 'java'
apply plugin: 'application'
apply plugin: 'eclipse'
apply plugin: 'io.quarkus'

group = 'dev.buesing'
sourceCompatibility = JavaVersion.VERSION_11
targetCompatibility = JavaVersion.VERSION_11

idea {
    module {
        inheritOutputDirs = false
        outputDir = compileJava.destinationDir
        testOutputDir = compileTestJava.destinationDir
    }
}

dependencies {

    implementation "javax.validation:validation-api:${validation_api_version}"

    implementation 'io.quarkus:quarkus-resteasy-server-common'
    implementation 'io.quarkus:quarkus-arc'
    implementation 'io.quarkus:quarkus-config-yaml'
    implementation 'io.quarkus:quarkus-hibernate-validator'
    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")

    // lombok dependencies
    compileOnly "org.projectlombok:lombok:${lombok_version}"
    annotationProcessor "org.projectlombok:lombok:${lombok_version}"

    testCompileOnly "org.projectlombok:lombok:${lombok_version}"
    testAnnotationProcessor "org.projectlombok:lombok:${lombok_version}"
}

// only one SL4J should be part of runtime classpath, logback.
configurations.all {
    exclude group: "org.slf4j", module: "slf4j-log4j12"
}
```

gradle.properties
```
quarkusPluginVersion=1.7.0.Final
quarkusPlatformArtifactId=quarkus-universe-bom
quarkusPlatformGroupId=io.quarkus
quarkusPlatformVersion=1.7.0.Final
org.gradle.logging.level=INFO
validation_api_version=2.0.1.Final
lombok_version=1.18.12
quarkus_plugin_version=1.7.0.Final
```

In addition I am not seeing Hibernate Validation working (linger has @Min(0) annotation, but that isn't being honored)

**Configuration**
```properties
cluster:
  bootstrap-servers: localhost:19092,localhost:29092,localhost:39092
  security-protocol: PLAINTEXT
  linger: -10
```

**Screenshots**

**Environment (please complete the following information):**

uname -a
Darwin Neils-MacBook.local 19.0.0 Darwin Kernel Version 19.0.0: Wed Oct 23 18:29:05 PDT 2019; root:xnu-6153.41.3~44/RELEASE_X86_64 x86_64

java -version
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)

Not using GraalVM yet

Quarkus 1.7.0.Final

./gradlew --version
Initialized native services in: /Users/buesing/.gradle/native

------------------------------------------------------------
Gradle 6.5
------------------------------------------------------------

Build time:   2020-06-02 20:46:21 UTC
Revision:     a27f41e4ae5e8a41ab9b19f8dd6d86d7b384dad4

Kotlin:       1.3.72
Groovy:       2.5.11
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          11.0.2 (Oracle Corporation 11.0.2+9)
OS:           Mac OS X 10.15.1 x86_64

**Additional context**
(Add any other context about the problem here.)

### LINKED ISSUE #11782: Quarkus EnvConfigSource behaves differently from SmallRye ConfigSource
Looking at the `EnvConfigSource` class in [ConfigUtils.java](https://github.com/quarkusio/quarkus/blob/master/core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java), which AFAICT is used at build-time, I notice that `getValue()` never checks for an exact match. Nor does it as a last resort "normalize" the sanitized property name (non-alphanumerics replaced by `_`) by converting it to upper-case.

Unless I am missing something this looks wrong to me, when I compare with the MicroProfile Config spec and also the SmallRye implementation class `EnvConfigSource`.

/cc @radcortez

### LINKED ISSUE #14847: Allow specifying optional config section with YAML configuration
**Describe the bug**
See [here](https://quarkusio.zulipchat.com/#narrow/stream/187030-sers/topic/Option.20configuration.20not.20working.3F)
**Expected behavior**
Optional configuration sections should be allowed with Quarkus YAML config.

**Actual behavior**
I know of no way of specifying optional section with YAML config.

**To Reproduce**
Unpack this project 
[issue_optional_config.zip](https://github.com/quarkusio/quarkus/files/5931906/issue_optional_config.zip)
and follow the instruction [here](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Option.20configuration.20not.20working.3F)

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes #9895
- Fixes #11475
- Fixes #11782

SmallRye Config 1.11.0 Release Notes:
https://github.com/smallrye/smallrye-config/releases/tag/1.11.0

### CODE REVIEW

File: core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * We don't really use this, because these are configurations for the config itself, so it causes a chicken / egg
+ * problem, but we have it so the configurations can be properly documented.
+ *
+ * Relocation of the Config configurations to the Quarkus namespace is done in ConfigUtils#configBuilder.
+ */
+@ConfigRoot(name = ConfigItem.PARENT, phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class ConfigConfig {
```
Comment (gsmet) [inline core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java:16]: Didn't see that one coming :)

Comment (radcortez) [reply core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java:16]: LOL!

Comment (radcortez) [reply core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java:16]: You got a better name? :P

Comment (gsmet) [reply core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java:16]: Nope :). I let you mark it ready for review once you think it's ready!

Comment (radcortez) [reply core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java:16]: Sure :)

File: core/runtime/src/test/java/io/quarkus/runtime/configuration/DotEnvTestCase.java [hunk 2]
```diff
@@ -83,7 +84,7 @@ public void testProperties() throws IOException {
                 singletonMap("FOO_BAR", "foo.bar"),
                 singletonMap("foo.baz", "nothing")));
         assertEquals("foo.bar", config.getValue("foo.bar", String.class));
-        assertFalse(config.getOptionalValue("foo.baz", String.class).isPresent());
+        assertTrue(config.getOptionalValue("foo.baz", String.class).isPresent());
```
Comment (gsmet) [inline core/runtime/src/test/java/io/quarkus/runtime/configuration/DotEnvTestCase.java:87]: I'm a bit puzzled by this change?

**Review by gsmet — COMMENTED** [review]
Added a small question inline.

File: core/runtime/src/test/java/io/quarkus/runtime/configuration/DotEnvTestCase.java (hunk 2 above)
Comment (radcortez) [reply core/runtime/src/test/java/io/quarkus/runtime/configuration/DotEnvTestCase.java:87]: Related with https://github.com/quarkusio/quarkus/issues/11782. If someone sets the config key with dot separators in the DotEnv source, we were only looking for the equivalent normalised to upper case and underscore separated. The SR Config DotEnv and Env source also search for an exact match.

**Review by gsmet — APPROVED** [review]

**Pull request merged**